techniques/applications/repoGpgKeyManagement/1.0/metadata.xml

```diff
@@ -0,0 +1,125 @@
+     in this Technique to import and parse variables in.
+     Note: the ".st" extension will be added automatically, don't specify it here -->
+  <TMLS>
+    <TML name="repoGpgKeyManagement"/>
+  </TMLS>
+
+  <!-- List of CFEngine bundles in the above .st TMLS to call.
+     These will be added to the CFEngine bundlesequence.
+       Technically, they don't have to have the same name as the
+     TML, but if there is one bundle per TML, it is recommended. -->
+  <BUNDLES>
+    <NAME>check_repo_gpg_key</NAME>
+  </BUNDLES>
+
+  <!-- Define agent and OS compatibility for this Technique.
+     Currently, this is for information purposes only. 
+     ## TODO: It has not been tested on any older OS, only on latest-greatest (SLES11, CentOS 7, Ubuntu 14)
+  <COMPATIBLE>
+    <OS version=">= 5">RHEL</OS>
+    <OS version=">= 11">SuSE Linux</OS>
+    <OS version=">= 4 (Etch)">Debian</OS>
+    <OS version=">= 5">RHEL / CentOS</OS>
+  </COMPATIBLE>
+-->
+
+  <TRACKINGVARIABLE>
+    <SAMESIZEAS>GPG_KEY_HASH</SAMESIZEAS>
+  </TRACKINGVARIABLE>
+
+  <!-- From here on, define variables to display in the web interface
+     All variables must be contained in a section.
+     Sections may be multivalued, or not -->
+ <SECTIONS>
+   <SECTION name="Repository GPG Key Management" multivalued="true" component="true" componentKey="GPG_KEY_HASH">
+     <SELECT1>
+       <NAME>GPG_KEY_ACTION</NAME>
+       <DESCRIPTION>Which operation should be done with this GPG Key</DESCRIPTION>
+       <ITEM>
+         <LABEL>Import (hash and content required)</LABEL>
+         <VALUE>add</VALUE>
+       </ITEM>
+       <ITEM>
+         <LABEL>Remove (only hash required)</LABEL>
+         <VALUE>del</VALUE>
+       </ITEM>
+       <CONSTRAINT>
+         <DEFAULT>add</DEFAULT>
+       </CONSTRAINT>
+     </SELECT1>
+     <INPUT>
+       <NAME>GPG_KEY_HASH</NAME>
+       <CONSTRAINT>
+         <MAYBEEMPTY>false</MAYBEEMPTY> 
+         <REGEX error="Exactly 16 hexadecimal characters required"><![CDATA[ [a-fA-F0-9]{16} ]]></REGEX>
+       </CONSTRAINT>
+       <DESCRIPTION>Long hash of the GPG Key</DESCRIPTION>
+       <LONGDESCRIPTION>
+        You get it by looking for a line like `pub   2048R/70AF9E8139DB7C82 2013-01-31' when using thae command `gpg --list-keys --keyid-format=long'.
+        From that, fill in the 16 hexa-chars part like `70AF9E8139DB7C82' in the example.
+       </LONGDESCRIPTION>
+     </INPUT>
+     <INPUT>
+       <NAME>GPG_KEY_NAME</NAME>
+       <CONSTRAINT>
+         <MAYBEEMPTY>true</MAYBEEMPTY> 
+         <REGEX error="No double quotation marks"><![CDATA[ [^"]* ]]></REGEX>
+       </CONSTRAINT>
+       <DESCRIPTION>Description</DESCRIPTION>
+       <LONGDESCRIPTION>This is only used to identify the Key in the GUI, it has no role in adding/removing the Key on the systems</LONGDESCRIPTION>
+     </INPUT>
+     <INPUT>
+       <NAME>GPG_KEY_CONTENT</NAME>
+       <CONSTRAINT>
+         <MAYBEEMPTY>true</MAYBEEMPTY> 
```

Comment:
Why is that optional ?
This seems to be mandatory for me as you really don't use the hash